### ISSUE TITLE
[BUG]MQTT to Home Assistant no longer working on 4.4.6

### ISSUE THREAD

**Issue by PaleFlyer** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#809]
**Describe the bug**
Updated to 4.4.6, and noticed HA was no longer getting/sending data to the controller

**To Reproduce**
Update controller, and reboot

**Expected behavior**
Communication to exist

**Screenshots**
![image](https://github.com/user-attachments/assets/d6d5f3b6-0ead-4b97-ba72-856c65025bf2)
![image](https://github.com/user-attachments/assets/ff855ca3-fdc6-4f9c-900c-1fd87a3b8308)
I've tried a variety of configs on the MQTT page, this is the latest. I have verified my password is correct using another MQTT device.

**Hardware (please complete the following information):**
![image](https://github.com/user-attachments/assets/9673aaf3-b775-4ac6-a4ff-07da343fc3ee)
using the "red" level shifter

**Software (please complete the following information):**
 - Version 4.4.6

**Additional context**

**Comment by PaleFlyer** [comment]
Mosquitto isn't even seeing it, unless it somehow is coming in on an IP I don't have in my network.

Mosquitto is showing a LOT of hits from a 172 address. My 2 VLANS are 192 and 10. And the ESP is on the 192 network.

**Comment by patatman** [comment]
Just to verify, I'm running the latest version 4.4.6 and I don't have any issues.
MQTT is working fine. So this seems to be a local problem.
Are you still having issues? Maybe try power-cycling the entire unit?
<img width="254" alt="image" src="https://github.com/user-attachments/assets/e2a9bd22-ab9c-400c-a5b8-e79f65d7d15d">

**Comment by PaleFlyer** [comment]
> Just to verify, I'm running the latest version 4.4.6 and I don't have any issues. MQTT is working fine. So this seems to be a local problem. Are you still having issues? Maybe try power-cycling the entire unit? <img alt="image" width="254" src="https://private-user-images.githubusercontent.com/15031001/382263836-e2a9bd22-ab9c-400c-a5b8-e79f65d7d15d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA0NjcxOTQsIm5iZiI6MTczMDQ2Njg5NCwicGF0aCI6Ii8xNTAzMTAwMS8zODIyNjM4MzYtZTJhOWJkMjItYWI5Yy00MDBjLWE1YjgtZTc5ZjY1ZDdkMTVkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTAxVDEzMTQ1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdjMmRkOTAzZDI1NzcxZDNjNjNkYjkzMTNiN2RkM2M1YjI2MDU2MWY0N2M2ZWIxZjZhY2ZmZjlkNzBjMGNmZjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.JzvRd3Zx6A8ZMkvXoSb2bsX2YmgS3Vo2tQbAvy5UkIo">

I did try soft-rebooting the device, but I have not tried pulling the power.

**Comment by visualapproach** [comment]
You need to update the file(system) also:
![image](https://github.com/user-attachments/assets/85a85433-87dc-44b1-af60-af41cb1a79a4)

**Comment by PaleFlyer** [comment]
Clearly VSCode, PlatformIO, and/or the ESP hates me. I can't push the FileSystem OTA... So I'm gonna have to crack the case open again... So I guess I need to find the waterproof USB I had gotten for this kind of thing, and install that finally.
![image](https://github.com/user-attachments/assets/8a5e5f3e-912b-47d2-9cde-e43741ed0685)

**Comment by visualapproach** [comment]
Don't use "upload fs OTA". Use "upload fs"

**Comment by PaleFlyer** [comment]
Tried that to, said it couldn't grab the serial port for the network upload

**Comment by PaleFlyer** [comment]
For some reason the upload non-OTA worked this time...

**Comment by visualapproach** [comment]
Nice

**Comment by PaleFlyer** [comment]
Getting a "Connection_Unauthorized" now, but I have a feeling that is due to my server, as it's been showing some signs of issues for a while now, and just rebooted HomeAssistant as I can't open the logs as the API call is failing in HomeAssistant....

**Comment by PaleFlyer** [comment]
Okay, for no apparent reason, the user I have on all my other MQTT devices just doesn't like the hot tub, but... I got it working.

Is there any way to WRITE the time to the timers?

IE with blowing out the file system, my counters on the filter, and the chlorine are all saying it's been over 2000 days...

The Chlorine and the filter rinse I need to do today normally, but the wash/replace is not a "today" item...

**Comment by visualapproach** [comment]
> Is there any way to WRITE the time to the timers?

No, but the next update you can just upload individual files instead of the whole file system. By choosing the UPLOAD FILE in the UI menu, and upload them one at a time. A bit of a hassle, but the saved data will be retained. Or you can download all the saved settings and put it in the data folder before uploading the FS.

**Comment by Moris101010** [comment]
I also have a problem with MQTT. I can enter a password consisting of 32 characters, but in HA I have a much longer one. How can I enter a long password?
There was no problem on the old version.

**Comment by visualapproach** [comment]
> I also have a problem with MQTT. I can enter a password consisting of 32 characters, but in HA I have a much longer one. How can I enter a long password?
> 
> There was no problem on the old version.

You can edit line 69 in data/mqtt.html and upload the file to the esp from the UI menu. There is a max total message length though so it will probably crash if all your fields are excessively long. But go ahead and try a larger maxlength value!
